internal/v1/server.go

```diff
@@ -408,8 +408,21 @@ func buildOSTreeOptions(ostreeOptions *OSTree) *cloudapi.OSTree {
 }
 
 func (h *Handlers) ComposeImage(ctx echo.Context) error {
+	idHeader, err := h.getIdentityHeader(ctx)
+	if err != nil {
+		return err
+	}
+
+	quotaOk, err := common.CheckQuota(idHeader.Identity.AccountNumber, h.server.db)
+	if err != nil {
+		return err
+	}
+	if !quotaOk {
+		return echo.NewHTTPError(http.StatusUnauthorized, "Quota exceeded for user")
```

Comment:
http codes are badly named >.> 

`http.StatusForbidden` is actually correct here since that's an authorization failure. `http.StatusUnauthorized` is an authentication failure. And keep in mind that any 401s for CRC would need to be 404s. But in this case a 403 is the correct one.

Reply:
I'll try to keep this in mind :sweat_smile: